vcstool/clients/git.py

```diff
@@ -49,21 +55,82 @@ def _branch(self, all):
     def branch(self, command):
         return self._branch(command.all)
 
-    def compare(self, command):
+    def _unstaged_changes(self):
+        result = self.custom(GitCustomCommand(['diff', '--no-ext-diff', '--quiet']))
+        return result['returncode'] == 1
+
+    def _staged_changes(self):
+        result = self.custom(GitCustomCommand(['diff', '--no-ext-diff', '--cached', '--quiet']))
+        return result['returncode'] == 1
+
+    def _untracked_files(self):
+        result = self.custom(GitCustomCommand(['ls-files', '--others', '--exclude-standard']))
+        return result['output'] != ''
+
+    def _stashes(self):
+        result = self.custom(GitCustomCommand(['rev-parse', '--verify', '--quiet', 'refs/stash']))
+        return result['returncode'] == 0
+
+    def _get_ahead_behind(self, local_branch, remote, remote_branch):
+        """Returns an (int, int) tuple corresponding to the (ahead, behind) commit numbers."""
+        if not remote or not remote_branch:
+            return (0, 0)
+        # remove leading "refs/heads/"
+        prefix = 'refs/heads/'
+        if remote_branch.startswith(prefix):
+            remote_branch = remote_branch[len(prefix):]
+        result = self.custom(GitCustomCommand(['rev-list', '--left-right', '--count',
+                                               f'{local_branch}...{remote}/{remote_branch}']))
+        # If the command failed, just return an error sentinel.
+        if result['returncode']:
+            return (-1, -1)
```

Comment:
Assuming the caller will check to see if ahead is non-zero and behind is non-zero to determine if the sentinel value has been set? I am not a big python guy, but should we return `None` instead?

Reply:
Yes I agree that is a more pythonic way to indicate this, thanks!